### PR TITLE
Feature: Add stop area for 3D printing

### PR DESCRIPTION
**Pull request by bbaa77770**
[LU-2029](https://snapmaker2.atlassian.net/browse/LU-2029)

### CODE REVIEW

File: src/app/flux/printing/index.js [hunk 1]
```diff
@@ -349,15 +355,99 @@ export const actions = {
...
+        }
+        if (helpersExtruderConfig.support === RIGHT_EXTRUDER_MAP_NUMBER) {
+            useRight = true;
+        } else {
+            useLeft = true;
+        }
+        modelGroup.getModels().forEach((model) => {
+            // TODO, use constants
+            if (model.type === 'baseModel' || model.type === 'group') {
+                if (model.extruderConfig.infill === RIGHT_EXTRUDER_MAP_NUMBER || model.extruderConfig.infill === BOTH_EXTRUDER_MAP_NUMBER) {
```
Comment (wen-qing-255) [inline src/app/flux/printing/index.js:385]: the `if` list maybe can simplify

File: src/app/flux/printing/index.js [hunk 2]
```diff
@@ -349,15 +355,99 @@ export const actions = {
...
+                    useRight = true;
+                }
+                if (model.extruderConfig.infill === LEFT_EXTRUDER_MAP_NUMBER || model.extruderConfig.infill === BOTH_EXTRUDER_MAP_NUMBER) {
+                    useLeft = true;
+                }
+                if (model.extruderConfig.shell === RIGHT_EXTRUDER_MAP_NUMBER || model.extruderConfig.infill === BOTH_EXTRUDER_MAP_NUMBER) {
```
Comment (wen-qing-255) [inline src/app/flux/printing/index.js:391]: `model.extruderConfig.infill === BOTH_EXTRUDER_MAP_NUMBER` is same as above

File: src/app/models/ModelGroup.js [hunk 3]
```diff
@@ -477,6 +477,7 @@ class ModelGroup extends EventEmitter {
 
     updateBoundingBox(bbox) {
         this._bbox = bbox;
+        this._checkAnyModelOversteppedOrSelected();
```
Comment (wen-qing-255) [inline src/app/models/ModelGroup.js:480]: just update platform boundingBox need this check?

File: src/app/ui/widgets/PrintingVisualizer/PrintableCube.jsx [hunk 4]
```diff
@@ -13,23 +13,106 @@ import Grid from '../../../three-extensions/objects/Grid';
...
 
     update = () => {
         this.dispatchEvent({ type: 'update' });
     };
 
+    _setupStopArea() {
+        const { left, right, front, back } = this.stopArea;
+        const { x, y } = this.size;
+        // front
+        const geometry1 = new PlaneGeometry(x, front);
```
Comment (xiekun1992) [inline src/app/ui/widgets/PrintingVisualizer/PrintableCube.jsx:64]: change name

**Review by xiekun1992 — APPROVED** [review]

**Pull request merged**